public/js/angular/directives/wysiwyg.js

```diff
@@ -0,0 +1,405 @@
+          type: 'h3'
+        }, {
+          name: loc.wysiwyg.HEADING_4,
+          type: 'h4'
+        }, {
+          name: loc.wysiwyg.HEADING_5,
+          type: 'h5'
+        }, {
+          name: loc.wysiwyg.HEADING_6,
+          type: 'h6'
+        }];
+
+        scope.setLayoutView = function(view) {
+          scope.wysiwyg.currentView = view;
+        };
+
+        scope.setElement = function(type) {
+          scope.formatAction('formatblock', type);
+        };
+
+        scope.getCurrentElement = function() {
+          var block = $document[0].queryCommandValue('formatblock');
+
+          for(var i = 0; i < scope.availableElements.length; i++) {
+            if(scope.availableElements[i].type === block) {
+              return scope.availableElements[i];
+            }
+          }
+
+          scope.setElement('p');
+          return scope.availableElements[0];
+        }
+
+        scope.formatAction = function(action, arguments) {
+          if(scope.wysiwyg.currentView !== 'editable') {
+            return;
+          }
+
+          $document[0].execCommand(action, false, arguments);
+        };
+
+        scope.isFormatActive = function(type) {
+          return $document[0].queryCommandState(type)
+        };
+
+        scope.clearStyles = function() {
+          scope.formatAction('removeFormat');
+          scope.setElement('p');
+        };
+
+        scope.showInsertLinkModal = function() {
+          if(!scope.layoutLink) {
+            scope.layoutLink = {
+              newTab: true
+            };
+          }
+
+          scope.layoutLink.text = scope.getSelection();
+          scope.saveSelection();
+          angular.element(element).find('[insert-link-modal]').modal('show');
+        };
+
+        scope.testLayoutLink = function() {
+          $window.open(scope.layoutLink.url, '_blank');
+        };
+
+        scope.insertLayoutLink = function() {
+          angular.element(element).find('[insert-link-modal]').modal('hide');
+          scope.restoreSelection();
+
+          var link = '<a href="' + scope.layoutLink.url + '" ' + (scope.layoutLink.newTab ? 'target="_blank"' : '') + '>' + scope.layoutLink.text + '</a>';
+          scope.formatAction('inserthtml', link);
+        };
+
+        scope.insertReadMore = function() {
+          scope.restoreSelection();
+          scope.formatAction('inserthtml', '<hr class="read_more_break"></hr>');
+        };
+
+        scope.mediaAvailable = function() {
+          if(scope.media && scope.media instanceof Array) {
+            return true;
+          }
+
+          return false;
+        };
+
+        scope.showInsertMediaModal = function() {
+          angular.element(element).find('[contenteditable]').focus();
+          scope.saveSelection();
+          angular.element(element).find('[insert-media-modal]').modal('show');
+        };
+
+        scope.associateMedia = function() {
+          angular.element(element).find('[insert-media-modal]').modal('hide');
```

Comment:
That's just an attribute selector in jquery and is normal syntax. We don't need to make them constants, IMO.